### ISSUE TITLE
getPassword unit test incorrectly uses popen3()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1963]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1963|
|Type|defect|
|Created|2006-07-27 22:49:58Z|

This needs to be spawnProcess instead.

This is killing the 2.5 buildslave.

<details><summary>Searchable metadata</summary>

```
trac-id__1963 1963
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1154040598000000 1154040598000000
changetime__1154279846000000 1154279846000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Waiting review in getpassword-1963

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

I'm beginning to review it

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez set owner to @exarkun|
|-|-|

The test is certainly cleaner with spawnProcess.
It is kinda sad that it is unix-only, it could probably be portable.
I think it should go in.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

This is 100 times clearer than the original test. Assuming the tests pass, my only thoughts are:

 * Maybe explain why 'unix only'
 * Maybe put a comment somewhere that explains the script prints the result of getPassword() to stdout. 

Make those changes if you'd like, then feel free to merge w/out bouncing back for review. Otherwise, merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17765]) Merge getpassword-1963

Author: exarkun
Reviewer: jml
Fixes [#1963](https://github.com/twisted/twisted/issues/1963)

Replace the usage of popen3 in the test for getPassword with spawnProcess.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
